Change API-DIFF Job to use default image

Change API-DIFF Job to use default image instead of our custom

With the 'centos-7' agent we are running into https://bugs.eclipse.org/bugs/show_bug.cgi?id=564559

When the centos-7 image is configured custom the pod is restarted in an endless loop:
```
08:53:43  Created Pod: xtext/xtext-xtend-api-diff-6bw3g-b3qzn
08:53:43  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Scheduled] Successfully assigned xtext-xtend-api-diff-6bw3g-b3qzn to osnode-16.eclipse.org
08:53:43  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "gradle" 
08:53:43  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "workspace-volume" 
08:53:43  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "volume-known-hosts" 
08:53:44  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "default-token-t46wf" 
08:53:46  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Pulled] Container image "docker.io/eclipsecbi/jiro-agent-centos-7:remoting-4.5" already present on machine
08:53:47  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Created] Created container
08:53:47  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Started] Started container
08:53:47  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Pulled] Container image "docker.io/eclipsecbi/jiro-agent-basic:remoting-4.5" already present on machine
08:53:48  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Created] Created container
08:53:48  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Started] Started container
08:53:48  xtext/xtext-xtend-api-diff-6bw3g-b3qzn Container centos-7 was terminated (Exit Code: 0, Reason: Completed)
08:53:49  [Normal][xtext/xtext-xtend-api-diff-6bw3g-b3qzn][Killing] Killing container with id docker://jnlp:Need to kill Pod
08:54:03  Created Pod: xtext/xtext-xtend-api-diff-6bw3g-z0rmp
08:54:03  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Scheduled] Successfully assigned xtext-xtend-api-diff-6bw3g-z0rmp to osnode-13.eclipse.org
08:54:03  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "gradle" 
08:54:03  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "workspace-volume" 
08:54:03  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "volume-known-hosts" 
08:54:03  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][SuccessfulMountVolume] MountVolume.SetUp succeeded for volume "default-token-t46wf" 
08:54:06  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Pulled] Container image "docker.io/eclipsecbi/jiro-agent-centos-7:remoting-4.5" already present on machine
08:54:07  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Created] Created container
08:54:07  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Started] Started container
08:54:07  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Pulled] Container image "docker.io/eclipsecbi/jiro-agent-basic:remoting-4.5" already present on machine
08:54:08  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Created] Created container
08:54:09  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Started] Started container
08:54:10  xtext/xtext-xtend-api-diff-6bw3g-z0rmp Container centos-7 was terminated (Exit Code: 0, Reason: Completed)
08:54:11  [Normal][xtext/xtext-xtend-api-diff-6bw3g-z0rmp][Killing] Killing container with id docker://jnlp:Need to kill Pod
```
Agent configuration was:
```
  agent {
    kubernetes {
      label 'xtext-xtend-api-diff'
      defaultContainer 'centos-7'
      yaml '''
apiVersion: v1
kind: Pod
spec:
  containers:
  - name: centos-7
    image: 'docker.io/eclipsecbi/jiro-agent-centos-7:remoting-4.5'
    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
    resources:
      limits:
        memory: "3.5Gi"
        cpu: "1.0"
      requests:
        memory: "3.5Gi"
        cpu: "1.0"
    volumeMounts:
    - name: gradle
      mountPath: /home/jenkins/.gradle
  volumes:
  - name: volume-known-hosts
    configMap:
      name: known-hosts
  - name: gradle
    emptyDir: {}
    '''
    }
  }
```


Using the jnlp image runs into a Java 11 issue when executing the diff:
```
05:37:14  Exception in thread "main" java.lang.NoClassDefFoundError: javax.xml.bind.SchemaOutputResolver
05:37:14  	at de.dhuebner.japicmp.FolderComparator.createReport(FolderComparator.java:78)
05:37:14  	at de.dhuebner.japicmp.CheckFolder.main(CheckFolder.java:44)
05:37:14  Caused by: java.lang.ClassNotFoundException: javax.xml.bind.SchemaOutputResolver
05:37:14  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:766)
05:37:14  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
05:37:14  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1083)
05:37:14  	... 2 more
```

Agent configuration was:
```
  agent {
    kubernetes {
      label 'xtext-xtend-api-diff'
      defaultContainer 'jnlp'
      yaml '''
apiVersion: v1
kind: Pod
spec:
  containers:
  - name: jnlp
    image: 'eclipsecbi/jenkins-jnlp-agent'
    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
    resources:
      limits:
        memory: "3.5Gi"
        cpu: "1.0"
      requests:
        memory: "3.5Gi"
        cpu: "1.0"
    volumeMounts:
    - name: gradle
      mountPath: /home/jenkins/.gradle
  volumes:
  - name: volume-known-hosts
    configMap:
      name: known-hosts
  - name: gradle
    emptyDir: {}
    '''
    }
  }
```

--- a/.gitignore
+++ b/.gitignore
@@ -15,3 +15,4 @@ hs_err_pid*
 tmf-xtext-Update*
 eclipse
 output
+japicmp.properties

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -1,76 +1,7 @@
 pipeline {
   agent {
     kubernetes {
-      label 'xtext-xtend-api-diff-' + env.BUILD_NUMBER
-      defaultContainer 'xtext-buildenv'
-      yaml '''
-apiVersion: v1
-kind: Pod
-spec:
-  containers:
-  - name: jnlp
-    image: 'eclipsecbi/jenkins-jnlp-agent'
-    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
-    volumeMounts:
-    - mountPath: /home/jenkins/.ssh
-      name: volume-known-hosts
-    resources:
-      limits:
-        memory: "0.5Gi"
-        cpu: "0.2"
-      requests:
-        memory: "0.5Gi"
-        cpu: "0.2"
-  - name: xtext-buildenv
-    image: docker.io/smoht/xtext-buildenv:0.7
-    tty: true
-    resources:
-      limits:
-        memory: "3.5Gi"
-        cpu: "1.0"
-      requests:
-        memory: "3.5Gi"
-        cpu: "1.0"
-    volumeMounts:
-    - name: settings-xml
-      mountPath: /home/jenkins/.m2/settings.xml
-      subPath: settings.xml
-      readOnly: true
-    - name: toolchains-xml
-      mountPath: /home/jenkins/.m2/toolchains.xml
-      subPath: toolchains.xml
-      readOnly: true
-    - name: settings-security-xml
-      mountPath: /home/jenkins/.m2/settings-security.xml
-      subPath: settings-security.xml
-      readOnly: true
-    - name: m2-repo
-      mountPath: /home/jenkins/.m2/repository
-  volumes:
-  - name: volume-known-hosts
-    configMap:
-      name: known-hosts
-  - name: settings-xml
-    secret:
-      secretName: m2-secret-dir
-      items:
-      - key: settings.xml
-        path: settings.xml
-  - name: toolchains-xml
-    configMap:
-      name: m2-dir
-      items:
-      - key: toolchains.xml
-        path: toolchains.xml
-  - name: settings-security-xml
-    secret:
-      secretName: m2-secret-dir
-      items:
-      - key: settings-security.xml
-        path: settings-security.xml
-  - name: m2-repo
-    emptyDir: {}
-    '''
+      label 'centos-7'
     }
   }
 

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -1,33 +1,7 @@
 pipeline {
   agent {
     kubernetes {
-      label 'xtext-xtend-api-diff'
-      defaultContainer 'jnlp'
-      yaml '''
-apiVersion: v1
-kind: Pod
-spec:
-  containers:
-  - name: jnlp
-    image: 'eclipsecbi/jenkins-jnlp-agent'
-    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
-    resources:
-      limits:
-        memory: "3.5Gi"
-        cpu: "1.0"
-      requests:
-        memory: "3.5Gi"
-        cpu: "1.0"
-    volumeMounts:
-    - name: gradle
-      mountPath: /home/jenkins/.gradle
-  volumes:
-  - name: volume-known-hosts
-    configMap:
-      name: known-hosts
-  - name: gradle
-    emptyDir: {}
-    '''
+      label 'centos-7'
     }
   }
 
@@ -35,6 +9,10 @@ spec:
     buildDiscarder(logRotator(numToKeepStr:'15'))
     disableConcurrentBuilds()
     timeout(time: 45, unit: 'MINUTES')
+  }
+
+  environment {
+      GRADLE_USER_HOME = "$WORKSPACE/.gradle" // workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=564559
   }
 
   // https://jenkins.io/doc/book/pipeline/syntax/#triggers
@@ -103,14 +81,12 @@ spec:
               color = '#666666'
           }
 
-          /*
           slackSend (
             message: "${lastResult} => ${curResult}: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>",
             botUser: true,
             channel: 'xtext-builds',
             color: "${color}"
           )
-          */
         }
       }
     }

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -1,7 +1,33 @@
 pipeline {
   agent {
     kubernetes {
-      label 'centos-7'
+      label 'xtext-xtend-api-diff'
+      defaultContainer 'jnlp'
+      yaml '''
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: jnlp
+    image: 'eclipsecbi/jenkins-jnlp-agent'
+    args: ['\$(JENKINS_SECRET)', '\$(JENKINS_NAME)']
+    resources:
+      limits:
+        memory: "3.5Gi"
+        cpu: "1.0"
+      requests:
+        memory: "3.5Gi"
+        cpu: "1.0"
+    volumeMounts:
+    - name: gradle
+      mountPath: /home/jenkins/.gradle
+  volumes:
+  - name: volume-known-hosts
+    configMap:
+      name: known-hosts
+  - name: gradle
+    emptyDir: {}
+    '''
     }
   }
 
@@ -25,14 +51,26 @@ pipeline {
     stage('Checkout') {
       steps {
         checkout scm
+        script {
+          currentBuild.displayName = "#${BUILD_NUMBER}(${params.OLD_VERSION} -> ${params.NEW_VERSION})"
+        }
+      }
+    }
+    stage('Build xtext-apidiff') {
+      steps {
+        script {
+          dir ('xtext-apidiff') {
+            sh '''
+              ./gradlew clean build
+              cp -f build/libs/japicmp-ext.jar ..
+            '''
+          }
+        }
       }
     }
     stage('Create API Diff') {
       steps {
-        script {
-          currentBuild.displayName = "#${BUILD_NUMBER}(${params.OLD_VERSION} -> ${params.NEW_VERSION})"
-        }
-        sh "./create-api-diff.sh"
+        sh './create-api-diff.sh'
       }
     }
   } // END stages
@@ -65,12 +103,14 @@ pipeline {
               color = '#666666'
           }
 
+          /*
           slackSend (
             message: "${lastResult} => ${curResult}: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>",
             botUser: true,
             channel: 'xtext-builds',
             color: "${color}"
           )
+          */
         }
       }
     }

--- a/create-api-diff.sh
+++ b/create-api-diff.sh
@@ -71,7 +71,7 @@ download () {
    ZIP_FILE=tmf-xtext-Update-$XTEXT_VERSION.zip
 
    if [ ! -d tmf-xtext-Update-$XTEXT_VERSION ]; then
-      echo "Downloading Xtext $XTEXT_VERSION"
+      echo "Downloading Xtext $XTEXT_VERSION from $DOWNLOAD_URL"
       # check existence of file
       if [ $(curl -s -o /dev/null -w '%{http_code}' $DOWNLOAD_URL) == 404 ]; then
          echo "Not found: $DOWNLOAD_URL"
@@ -88,9 +88,7 @@ if [ -d tmf-xtext-Update-$DEV_VERSION ]; then
   rm -r tmf-xtext-Update-$DEV_VERSION
 fi
 DOWNLOAD_URL=https://ci.eclipse.org/xtext/job/xtext-umbrella/job/master/lastStableBuild/artifact/build/org.eclipse.xtext.sdk.p2-repository-$DEV_VERSION-SNAPSHOT.zip
-
 download $DEV_VERSION $DOWNLOAD_URL
-
 
 # download NEW_VERSION and OLD_VERSION if not present
 for ((idx=0; idx<${#BUILD_IDS[@]}; ++idx)); do
@@ -99,7 +97,7 @@ for ((idx=0; idx<${#BUILD_IDS[@]}; ++idx)); do
 
    # only download relevant versions
    if [ $VERSION == $NEW_VERSION ] || [ $VERSION == $OLD_VERSION ]; then
-      download $VERSION $DOWNLOAD_URL
+      download $VERSION "https://download.eclipse.org/modeling/tmf/xtext/downloads/drops/$VERSION/$BUILD_ID/tmf-xtext-Update-$VERSION.zip"
    fi
 done
 

--- a/create-api-diff.sh
+++ b/create-api-diff.sh
@@ -2,7 +2,6 @@
 # Use the build container for testing to avoid OS specific interpretation:
 # docker run -it -v $(pwd):/xtext -w /xtext eclipsecbi/jiro-agent-centos-7 bash
 
-
 declare -A VERSION_2_BUILDID
 VERSION_2_BUILDID["2.25.0"]="R202103011429"
 VERSION_2_BUILDID["2.24.0"]="R202011301016"
@@ -108,10 +107,6 @@ done
 
 echo ""
 echo "Creating the actual diff"
-cd xtext-apidiff
-./gradlew clean build
-cd ..
-cp -f xtext-apidiff/build/libs/japicmp-ext.jar .
 
 # update apicmp.properties
 {

--- a/create-api-diff.sh
+++ b/create-api-diff.sh
@@ -1,121 +1,117 @@
 #!/bin/bash
+# Use the build container for testing to avoid OS specific interpretation:
+# docker run -it -v $(pwd):/xtext -w /xtext eclipsecbi/jiro-agent-centos-7 bash
 
-cd xtext-apidiff
-./gradlew clean build
-cd ..
+
+declare -A VERSION_2_BUILDID
+VERSION_2_BUILDID["2.25.0"]="R202103011429"
+VERSION_2_BUILDID["2.24.0"]="R202011301016"
+VERSION_2_BUILDID["2.23.0"]="R202008310926"
+VERSION_2_BUILDID["2.22.0"]="R202006021533"
+VERSION_2_BUILDID["2.21.0"]="R202003021509"
+VERSION_2_BUILDID["2.20.0"]="R201912021256"
+VERSION_2_BUILDID["2.19.0"]="R201909021322"
+VERSION_2_BUILDID["2.18.0"]="R201906031516"
+VERSION_2_BUILDID["2.17.1"]="R201904030733"
+VERSION_2_BUILDID["2.17.0"]="R201903041445"
+
+VERSIONS=(2.25.0 2.24.0 2.23.0 2.22.0 2.21.0 2.20.0 2.19.0 2.18.0 2.17.1 2.7.0)
+BUILD_IDS=(R202103011429\
+ R202011301016 \
+ R202008310926 \
+ R202006021533 \
+ R202003021509 \
+ R201912021256 \
+ R201909021322 \
+ R201906031516 \
+ R201904030733 \
+ R201903041445)
+
+DEV_VERSION=$(curl -s https://raw.githubusercontent.com/eclipse/xtext-core/master/gradle/versions.gradle |grep -Po "version = \'\K([^\']*)(?=\')" |sed 's/-SNAPSHOT//')
+
+# The Eclipse release to use
+ECLIPSE_RELEASE=2020-06
+ECLIPSE_TARGZ_FILE=eclipse-SDK-4.13-linux-gtk-x86_64.tar.gz
+ECLIPSE_TARGZ_DOWNLOAD_URL=http://download.eclipse.org/eclipse/downloads/drops4/R-4.13-201909161045/$ECLIPSE_TARGZ_FILE
+ECLIPSE_XTEXT_VERSION=${VERSIONS[0]}
+
+
 
 if [ -z "$NEW_VERSION" ]; then
   # if not set in environment use default
-  NEW_VERSION=2.26.0
+  NEW_VERSION=$DEV_VERSION
 fi
 
 if [ -z "$OLD_VERSION" ]; then
   # if not set in environment use default
-  OLD_VERSION=2.25.0
+  OLD_VERSION=${VERSIONS[0]}
 fi
 
 echo "Diffing $NEW_VERSION against $OLD_VERSION"
+echo ""
+echo "Current Xtext dev version is $DEV_VERSION"
 
-if [ -f japicmp-ext.jar ];
-then
-  rm japicmp-ext.jar
-fi
-cp xtext-apidiff/build/libs/japicmp-ext.jar .
-
-if [ ! -d eclipse ];
-then
+if [ ! -d eclipse ]; then
    echo "Downloading Eclipse"
    # Get basic Eclipse SDK
-   curl -m 1200 -sL 'http://download.eclipse.org/eclipse/downloads/drops4/R-4.13-201909161045/eclipse-SDK-4.13-linux-gtk-x86_64.tar.gz' --output 'eclipse-SDK-4.13-linux-gtk-x86_64.tar.gz' && tar -zxf eclipse-SDK-4.13-linux-gtk-x86_64.tar.gz
-   rm eclipse-SDK-4.13-linux-gtk-x86_64.tar.gz
-   # Install additional features: Xtext and dependent
+   curl -m 1200 -sL $ECLIPSE_TARGZ_DOWNLOAD_URL --output $ECLIPSE_TARGZ_FILE && tar -zxf $ECLIPSE_TARGZ_FILE
+   rm $ECLIPSE_TARGZ_FILE
+
+   echo "Installing additional features: Xtext and dependent"
    eclipse/eclipse -data eclipse/.director-ws -consolelog -noSplash -clean \
    -application org.eclipse.equinox.p2.director \
-   -metadataRepository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.25.0/,https://download.eclipse.org/modeling/emft/mwe/updates/releases/2.12.1/,http://download.eclipse.org/releases/2020-06,http://download.eclipse.org/lsp4j/updates/releases/,https://download.eclipse.org/tools/orbit/downloads/2020-09 \
-   -artifactRepository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.25.0/,https://download.eclipse.org/modeling/emft/mwe/updates/releases/2.12.1/,http://download.eclipse.org/releases/2020-06,http://download.eclipse.org/lsp4j/updates/releases/,https://download.eclipse.org/tools/orbit/downloads/2020-09 \
+   -metadataRepository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/$ECLIPSE_XTEXT_VERSION,https://download.eclipse.org/modeling/emft/mwe/updates/releases,http://download.eclipse.org/releases/$ECLIPSE_RELEASE,http://download.eclipse.org/lsp4j/updates/releases/,https://download.eclipse.org/tools/orbit/downloads/$ECLIPSE_RELEASE \
+   -artifactRepository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/$ECLIPSE_XTEXT_VERSION,https://download.eclipse.org/modeling/emft/mwe/updates/releases,http://download.eclipse.org/releases/$ECLIPSE_RELEASE,http://download.eclipse.org/lsp4j/updates/releases/,https://download.eclipse.org/tools/orbit/downloads/$ECLIPSE_RELEASE \
    -installIU org.eclipse.xtext.sdk.feature.group,org.eclipse.lsp4j.sdk.feature.group,org.eclipse.m2e.core,org.eclipse.buildship.core,org.kohsuke.args4j \
    -destination eclipse
 fi
 
-if [ ! -d tmf-xtext-Update-2.17.0 ];
-then
-   echo "Downloading Xtext 2.17.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.17.0/R201903041445/tmf-xtext-Update-2.17.0.zip' --output 'tmf-xtext-Update-2.17.0.zip' && unzip -q tmf-xtext-Update-2.17.0.zip -d tmf-xtext-Update-2.17.0
-   rm tmf-xtext-Update-2.17.0.zip
-fi
+download () {
+   XTEXT_VERSION=$1
+   DOWNLOAD_URL=$2
+   ZIP_FILE=tmf-xtext-Update-$XTEXT_VERSION.zip
 
-if [ ! -d tmf-xtext-Update-2.17.1 ];
-then
-   echo "Downloading Xtext 2.17.1"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.17.1/R201904030733/tmf-xtext-Update-2.17.1.zip' --output 'tmf-xtext-Update-2.17.1.zip' && unzip -q tmf-xtext-Update-2.17.1.zip -d tmf-xtext-Update-2.17.1
-   rm tmf-xtext-Update-2.17.1.zip
-fi
+   if [ ! -d tmf-xtext-Update-$XTEXT_VERSION ]; then
+      echo "Downloading Xtext $XTEXT_VERSION"
+      # check existence of file
+      if [ $(curl -s -o /dev/null -w '%{http_code}' $DOWNLOAD_URL) == 404 ]; then
+         echo "Not found: $DOWNLOAD_URL"
+         exit 1
+      fi
+      curl -m 1200 -sL $DOWNLOAD_URL --output $ZIP_FILE
+      unzip -q $ZIP_FILE -d tmf-xtext-Update-$XTEXT_VERSION
+      rm $ZIP_FILE
+   fi
+}
 
-if [ ! -d tmf-xtext-Update-2.18.0 ];
-then
-   echo "Downloading Xtext 2.18.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.18.0/R201906031516/tmf-xtext-Update-2.18.0.zip' --output 'tmf-xtext-Update-2.18.0.zip' && unzip -q tmf-xtext-Update-2.18.0.zip -d tmf-xtext-Update-2.18.0
-   rm tmf-xtext-Update-2.18.0.zip
+# Download latest Xtext build
+if [ -d tmf-xtext-Update-$DEV_VERSION ]; then
+  rm -r tmf-xtext-Update-$DEV_VERSION
 fi
+DOWNLOAD_URL=https://ci.eclipse.org/xtext/job/xtext-umbrella/job/master/lastStableBuild/artifact/build/org.eclipse.xtext.sdk.p2-repository-$DEV_VERSION-SNAPSHOT.zip
 
-if [ ! -d tmf-xtext-Update-2.19.0 ];
-then
-   echo "Downloading Xtext 2.19.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.19.0/R201909021322/tmf-xtext-Update-2.19.0.zip' --output 'tmf-xtext-Update-2.19.0.zip' && unzip -q tmf-xtext-Update-2.19.0.zip -d tmf-xtext-Update-2.19.0
-   rm tmf-xtext-Update-2.19.0.zip
-fi
+download $DEV_VERSION $DOWNLOAD_URL
 
-if [ ! -d tmf-xtext-Update-2.20.0 ];
-then
-   echo "Downloading Xtext 2.20.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.20.0/R201912021256/tmf-xtext-Update-2.20.0.zip' --output 'tmf-xtext-Update-2.20.0.zip' && unzip -q tmf-xtext-Update-2.20.0.zip -d tmf-xtext-Update-2.20.0
-   rm tmf-xtext-Update-2.20.0.zip
-fi
 
-if [ ! -d tmf-xtext-Update-2.21.0 ];
-then
-   echo "Downloading Xtext 2.21.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.21.0/R202003021509/tmf-xtext-Update-2.21.0.zip' --output 'tmf-xtext-Update-2.21.0.zip' && unzip -q tmf-xtext-Update-2.21.0.zip -d tmf-xtext-Update-2.21.0
-   rm tmf-xtext-Update-2.21.0.zip
-fi
+# download NEW_VERSION and OLD_VERSION if not present
+for ((idx=0; idx<${#BUILD_IDS[@]}; ++idx)); do
+   VERSION=${VERSIONS[idx]}
+   BUILD_ID=${BUILD_IDS[idx]}
 
-if [ ! -d tmf-xtext-Update-2.22.0 ];
-then
-   echo "Downloading Xtext 2.22.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.22.0/R202006021533/tmf-xtext-Update-2.22.0.zip' --output 'tmf-xtext-Update-2.22.0.zip' && unzip -q tmf-xtext-Update-2.22.0.zip -d tmf-xtext-Update-2.22.0
-   rm tmf-xtext-Update-2.22.0.zip
-fi
+   # only download relevant versions
+   if [ $VERSION == $NEW_VERSION ] || [ $VERSION == $OLD_VERSION ]; then
+      download $VERSION $DOWNLOAD_URL
+   fi
+done
 
-if [ ! -d tmf-xtext-Update-2.23.0 ];
-then
-   echo "Downloading Xtext 2.23.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.23.0/R202008310926/tmf-xtext-Update-2.23.0.zip' --output 'tmf-xtext-Update-2.23.0.zip' && unzip -q tmf-xtext-Update-2.23.0.zip -d tmf-xtext-Update-2.23.0
-   rm tmf-xtext-Update-2.23.0.zip
-fi
 
-if [ ! -d tmf-xtext-Update-2.24.0 ];
-then
-   echo "Downloading Xtext 2.24.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.24.0/R202011301016/tmf-xtext-Update-2.24.0.zip' --output 'tmf-xtext-Update-2.24.0.zip' && unzip -q tmf-xtext-Update-2.24.0.zip -d tmf-xtext-Update-2.24.0
-   rm tmf-xtext-Update-2.24.0.zip
-fi
 
-if [ ! -d tmf-xtext-Update-2.25.0 ];
-then
-   echo "Downloading Xtext 2.25.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.25.0/R202103011429/tmf-xtext-Update-2.25.0.zip' --output 'tmf-xtext-Update-2.25.0.zip' && unzip -q tmf-xtext-Update-2.25.0.zip -d tmf-xtext-Update-2.25.0
-   rm tmf-xtext-Update-2.25.0.zip
-fi
-
-if [ -d tmf-xtext-Update-2.26.0 ];
-then
-  rm -r tmf-xtext-Update-2.26.0
-fi
-echo "Downloading Xtext 2.26.0"
-# TODO make this better and faster. am not sure if we can guess the nightly number. or ....
-# ideally download official nightly
-curl -m 1200 --retry 5 -sL 'https://ci.eclipse.org/xtext/job/xtext-umbrella/job/master/lastStableBuild/artifact/build/org.eclipse.xtext.sdk.p2-repository-2.26.0-SNAPSHOT.zip' --output 'tmf-xtext-Update-2.26.0.zip' && unzip -q tmf-xtext-Update-2.26.0.zip -d tmf-xtext-Update-2.26.0
-rm tmf-xtext-Update-2.26.0.zip
+echo ""
+echo "Creating the actual diff"
+cd xtext-apidiff
+./gradlew clean build
+cd ..
+cp -f xtext-apidiff/build/libs/japicmp-ext.jar .
 
 # update apicmp.properties
 {
@@ -127,9 +123,5 @@ rm tmf-xtext-Update-2.26.0.zip
 
 } > japicmp.properties
 
-if [ -d output ];
-then
-  rm -r output
-fi
-echo "creating the actual diff"
+rm -rf output
 java -jar japicmp-ext.jar

--- a/xtext-apidiff/.classpath
+++ b/xtext-apidiff/.classpath
@@ -12,7 +12,13 @@
 			<attribute name="gradle_used_by_scope" value="main,test"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+	<classpathentry kind="src" output="bin/main" path="resources">
+		<attributes>
+			<attribute name="gradle_scope" value="main"/>
+			<attribute name="gradle_used_by_scope" value="main,test"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11/"/>
 	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
 	<classpathentry kind="output" path="bin/default"/>
 </classpath>

--- a/xtext-apidiff/.settings/org.eclipse.jdt.core.prefs
+++ b/xtext-apidiff/.settings/org.eclipse.jdt.core.prefs
@@ -1,11 +1,11 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
 org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
-org.eclipse.jdt.core.compiler.compliance=1.8
+org.eclipse.jdt.core.compiler.compliance=11
 org.eclipse.jdt.core.compiler.debug.lineNumber=generate
 org.eclipse.jdt.core.compiler.debug.localVariable=generate
 org.eclipse.jdt.core.compiler.debug.sourceFile=generate
 org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
-org.eclipse.jdt.core.compiler.source=1.8
+org.eclipse.jdt.core.compiler.source=11

--- a/xtext-apidiff/build.gradle
+++ b/xtext-apidiff/build.gradle
@@ -24,11 +24,15 @@ tasks.withType(JavaCompile) {
 	options.encoding = 'ISO-8859-1'
 }
 
-sourceCompatibility = '1.8'
+sourceCompatibility = '11'
 
 dependencies {
 	implementation "org.eclipse.xtend:org.eclipse.xtend.lib:$xtextVersion"
 	implementation 'com.github.siom79.japicmp:japicmp:0.5.0'
+	implementation 'javax.xml.bind:jaxb-api:2.2.7'
+	implementation 'com.sun.xml.bind:jaxb-core:2.2.7'
+	implementation 'com.sun.xml.bind:jaxb-impl:2.2.7'
+	implementation 'javax.activation:activation:1.1'
 }
 
 sourceSets {

--- a/xtext-apidiff/gradle/wrapper/gradle-wrapper.properties
+++ b/xtext-apidiff/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

